### PR TITLE
amp pages are requested via amp.guardian.com or ?amp=1 query parameters

### PR DESCRIPTION
**Pull request by TBonnin**
cc @NataliaLKB

### CODE REVIEW

**Comment by NataliaLKB** [comment]
:star2:

**Comment by NataliaLKB** [comment]
Asked @OliverJAsh to review

File: dev-build/conf/routes [hunk 1]
```diff
@@ -395,7 +395,6 @@ GET            /$path<[^/]+(/[^/]+)?>
 
 GET            /$path<[\w\d-]*(/[\w\d-]*)?/gallery/.*>/lightbox.json                                                             controllers.GalleryController.lightboxJson(path)
 GET            /$path<[\w\d-]*(/[\w\d-]*)?/gallery/.*>.json                                                                      controllers.GalleryController.renderJson(path)
-GET            /$path<[\w\d-]*(/[\w\d-]*)?/gallery/.*>/amp                                                                       controllers.GalleryController.render(path)
```
Comment (OliverJAsh) [inline dev-build/conf/routes]: Does this route also need to be removed in the non-dev-build `routes`?

File: common/app/implicits/Requests.scala [hunk 2]
```diff
@@ -22,14 +22,13 @@ trait Requests {
 
     lazy val isRss: Boolean = r.path.endsWith("/rss")
 
-    lazy val isAmp: Boolean = r.path.endsWith(AMP_SUFFIX)
+    lazy val isAmp: Boolean = r.getQueryString("amp").isDefined || r.host == Configuration.amp.host
```
Comment (OliverJAsh) [inline common/app/implicits/Requests.scala]: `isAmp` is true if the amp query param is defined and equal to `1`? Not sure it matters though…

**Comment by OliverJAsh** [comment]
This looks OK to me, but it's changing some parts of the system which I'm not completely familiar with. Had a good look over it though :-) :+1: 

Build is failing…

**Comment by NataliaLKB** [comment]
@OliverJAsh Thanks for taking a look!

We should have fixed the build now.

File: dev-build/conf/routes (hunk 1 above)
Comment (NataliaLKB) [inline dev-build/conf/routes]: This is seems to be the only place this is referenced. It was probably left over from a few months ago when we thought about galleries on amp. That won't happen now for a while.

**Pull request merged**